app/models/game.rb

```diff
@@ -53,30 +51,29 @@ def update()
         self.round +=1
         self.next_round = self.next_round + (30*60)
         self.save
+        #First update the income levels
+        update_income_levels()
       end
     end
     return self
   end
 
   def update_income_levels()
     round = self.round
-    # PR > 4, change Income +1
-    # PR < -1 and PR < -3, change Income -1
-    # PR < -3, change Income -2
+    
     teams = Team.all_without_incomes
     teams.each do |team|
-      pr = self.public_relations.where(round: round).where(team: team).sum(:pr_amount)
-      current_income = self.incomes.where(round: round, team: team).sum(:amount)
-
-      if pr >= 4
-        current_income += 1
-      elsif pr <=-1 and pr >=-3
-        current_income += -1
-      elsif pr < -3
-        current_income += -2
+      turn_0 = calculate_income_level(self.public_relations.where(round: 0).where(team: team).sum(:pr_amount) || 0)
+      if (round > 1)
+        amount = self.public_relations.where('round < ? and round > 0', [round - 1, 0].max).where(team: team).group(:round).sum(:pr_amount).reduce(6 + turn_0) do |memo, arr|
```

Comment:
We are updating income on the UI to only show income for the current round.  As such, this query only needs to reflect public relations of current_round - 1.